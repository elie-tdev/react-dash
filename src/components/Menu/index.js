export { default as MobileMenu } from './MobileMenu';
export { default as MobileMenuItem } from './MobileMenuItem';
