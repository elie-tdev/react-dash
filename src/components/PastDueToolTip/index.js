export { default as PastDueToolTip } from './PastDueToolTip';
