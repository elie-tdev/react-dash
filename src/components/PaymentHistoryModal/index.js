export { default as PaymentHistoryModal } from './PaymentHistoryModal';
