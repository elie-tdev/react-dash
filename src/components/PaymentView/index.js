export { default as PaymentView } from './PaymentView';
export { default as PaymentCurrent } from './PaymentCurrent';
export { default as PaymentPastDue } from './PaymentPastDue';
export { default as ScheduledPayment } from './ScheduledPayment';
export { default as NextPaymentView } from './NextPaymentView';
export { default as EnrollInAutoPay } from './EnrollInAutoPay';
export { default as NextPaymentCard } from './NextPaymentCard';
export { default as PastDueCard } from './PastDueCard';
export { default as AutoPayCard } from './AutoPayCard';
