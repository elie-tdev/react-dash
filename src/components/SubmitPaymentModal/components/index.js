export { default as CustomListItem } from './CustomListItem';
export { default as NewPaymentAccount } from './NewPaymentAccount';
export { default as EditPaymentAccount } from './EditPaymentAccount';
