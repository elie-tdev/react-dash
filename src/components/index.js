export * from './LoanGlance';
export * from './Button';
export * from './Header';
export * from './Footer';
export * from './LabelValueColumn';
export * from './LabelValueRow';
export * from './Loader';
export * from './CenteredScreenContent';
export * from './PaymentView';
export * from './Pagination';
export * from './Screen';
export * from './LoanSelect';
export * from './StatementGlance';
export * from './Menu';
export * from './PaymentHistoryModal';
export * from './PaymentStatus';
export * from './PaymentType';
export * from './SubmitPaymentModal';
export * from './GridContainer';
export * from './PastDueToolTip';
