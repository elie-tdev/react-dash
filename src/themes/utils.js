export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
