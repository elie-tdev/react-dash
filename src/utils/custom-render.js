import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
