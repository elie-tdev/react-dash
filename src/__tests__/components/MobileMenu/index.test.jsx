import React from 'react';
import merge from 'deepmerge';

import { render, cleanup, fireEvent } from '@/utils/custom-render';
import { MobileMenu } from '@/components';
import { initializeUserMock, MOCK_BANK_ACCOUNTS } from '@/fixtures/test_data';

jest.mock('@/customHooks/useBankAccounts', () => () => MOCK_BANK_ACCOUNTS);

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('MobileMenu', () => {
  afterEach(() => {
    cleanup();
  });

  test("renders a capitalized greeting with the user's first name if they have one", async () => {
    const { container, findByText } = render(<MobileMenu />, {
      menuContextProviderProps: {
        isMenuOpen: true,
        routeParams: { loanNumber: '337720' },
      },
      mocks: [initializeUserMock],
    });

    expect(container).toMatchSnapshot();
    expect(await findByText('Welcome back, Rick!')).toBeVisible();
  });

  test("renders a capitalized greeting with the user's first name even if they have multiple first names", async () => {
    const { container, findByText } = render(<MobileMenu />, {
      menuContextProviderProps: {
        isMenuOpen: true,
        routeParams: { loanNumber: '337720' },
      },
      mocks: [
        merge(initializeUserMock, {
          result: {
            data: {
              Contacts_Get: {
                payload: {
                  data: {
                    Firstname1: 'RICKY ticky',
                  },
                },
              },
            },
          },
        }),
      ],
    });

    expect(container).toMatchSnapshot();
    expect(await findByText('Welcome back, Ricky Ticky!')).toBeVisible();
  });

  test('renders a generalized greeting if the user has no first name', async () => {
    const { container, findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true },
    });

    expect(container).toMatchSnapshot();
    expect(await findByText('Welcome back!')).toBeVisible();
  });

  test('clicking on the preferences link takes the user to the preferences page', async () => {
    const { findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true, routeParams: { loanNumber: '337720' } },
    });

    fireEvent.click(await findByText('Preferences'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/dashboard/loans/337720/preferences');
  });

  test('clicking on the loanoverview link takes the user to the loanoverview page', async () => {
    const { findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true, routeParams: { loanNumber: '337720' } },
    });
    fireEvent.click(await findByText('Loan Overview'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/dashboard/loans/337720');
  });

  test('clicking on the Balance Detail link takes the user to the Balance Detail page', async () => {
    const { findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true, routeParams: { loanNumber: '337720' } },
    });
    fireEvent.click(await findByText('Balance Detail'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/dashboard/loans/337720/balance');
  });

  test('clicking on the Payment History link takes the user to the Payment History page', async () => {
    const { findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true, routeParams: { loanNumber: '337720' } },
    });
    fireEvent.click(await findByText('Payment History'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/dashboard/loans/337720/payments');
  });

  test('clicking on the Statements link takes the user to the Statements page', async () => {
    const { findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true, routeParams: { loanNumber: '337720' } },
    });
    fireEvent.click(await findByText('Statements'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/dashboard/loans/337720/statements');
  });

  test('clicking on the Help Center link takes the user to the Help Center page', async () => {
    const { findByText } = render(<MobileMenu />, {
      menuContextProviderProps: { isMenuOpen: true, routeParams: { loanNumber: '337720' } },
    });
    fireEvent.click(await findByText('Help Center'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/dashboard/loans/337720/helpCenter');
  });
});
