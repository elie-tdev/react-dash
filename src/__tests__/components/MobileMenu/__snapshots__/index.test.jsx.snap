// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu renders a capitalized greeting with the user's first name even if they have multiple first names 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`MobileMenu renders a capitalized greeting with the user's first name if they have one 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`MobileMenu renders a generalized greeting if the user has no first name 1`] = `
<div
  aria-hidden="true"
/>
`;
